src/extension/background-script/actions/payments/listByAccount.ts

```diff
@@ -1,24 +1,71 @@
-import type { MessagePaymentListByAccount } from "~/types";
+import { ConnectorTransaction } from "~/extension/background-script/connectors/connector.interface";
+import state from "~/extension/background-script/state";
+import type {
+  DbPayment,
+  MessagePaymentListByAccount,
+  Transaction,
+} from "~/types";
 
 import db from "../../db";
 
+// @Todo: renmae, due to fact that "byAccount" will be default
+
 const listByAccount = async (message: MessagePaymentListByAccount) => {
-  const { accountId } = message.args;
+  const accountId = state.getState().currentAccountId;
+
   // TODO: Add pagination instead of limiting to 2121
   const limit = message?.args?.limit || 2121;
 
-  const payments = await db.payments
+  const connector = await state.getState().getConnector();
+
+  // get IndexDB payments
+  const dbPayments = await db.payments
     .toCollection()
     .filter((p) => p.accountId === accountId)
     .limit(limit)
     .reverse()
     .sortBy("createdAt");
 
-  return {
-    data: {
-      payments,
-    },
-  };
+  try {
+    const result = await connector.getTransactions();
+
+    // map IndexDB payment data onto ConnectorTransaction
+    const transactions: Transaction[] = result.data.transactions.map(
+      (transaction: ConnectorTransaction) => {
+        const dbPayment = dbPayments.find(
+          (p: DbPayment) => p.preimage === transaction.payment_preimage
+        );
+
+        // @Todo: map transactions correctly
+        // @Todo: how to we use/map "title" and "description" and "memo"
+
+        // @Todo: this comes from convertPaymentToTransaction function - remove there if this is right
+        const title = dbPayment ? dbPayment.name || dbPayment.description : "";
+        return {
+          ...(dbPayment ? dbPayment : {}),
+          id: "",
```

Comment:
I guess the id should come from the transaction if it exists, otherwise use the dbPayment id

Reply:
Transactions don't have ids I think. I explicitly use dbPayment.id now. I think the original issue was `DbPayment` has id optional and `Transaction` not. Maybe this can be simplified if it turns out the interfaces can be optimized.